### PR TITLE
feat(pie-monorepo): DSW-2507 add Bundlewatch to repo and add CI checks

### PR DESCRIPTION
**Pull request by jamieomaguire**
### Adds the `bundlewatch` package and:
- Integrates with CI to generate HTML reports and fail a PR check if a component exceeds its max size limit
- Can be run locally
- Should create size histories when merged to main
- Updated the component generator to add a max size limit for new components with a baseline of `3kb`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -180,3 +180,7 @@ custom-elements.json
 # Yalc
 .yalc
 yalc.lock
+
+# Compsizer
+**/compsizer-failure-report.json
+**/component-bundle-sizes.json
```
Comment (ashleynolan) [inline .gitignore:186]: Is this to do with Compsizer rather than bundlewatch? (so may be able to remove)

File: package.json [hunk 2]
```diff
@@ -93,6 +94,7 @@
     "@vitest/coverage-c8": "0.29.8",
     "autoprefixer": "10.4.20",
     "babel-loader": "8.3.0",
+    "bundlewatch": "^0.4.0",
```
Comment (ashleynolan) [inline package.json:97]:
```suggestion
    "bundlewatch": "0.4.0",
```

**Review by ashleynolan — COMMENTED** [review]
Looks great – few comments but LGTM (assuming others are happy with approving the PoC outcome)

File: bundlewatch.config.json [hunk 3]
```diff
@@ -0,0 +1,97 @@
+{
+  "files": [
+    {
+      "path": "./packages/components/pie-assistive-text/dist/*.js",
+      "maxSize": "2 KB"
```
[the remark this answers is not in the document]
Comment (jamieomaguire) [reply bundlewatch.config.json:5]: regarding this ^ that `233B` is actually the size of the React file. The component itself is `1.1KB`. Shall we have separate paths for React and Index.js?
Screenshot 2024-11-06 at 17 41 18

[the remark this answers is not in the document]
Comment (jamieomaguire) [reply bundlewatch.config.json]: I will push an update that adds ~25% to each file size but also differentiates between index.js and react.js

File: package.json (hunk 2 above)
Comment (jamieomaguire) [reply package.json]: pushed 👍

**Review by raoufswe — APPROVED** [review]

File: bundlewatch.config.json [hunk 4]
```diff
@@ -0,0 +1,189 @@
+{
+  "files": [
+    {
+      "path": "./packages/components/pie-assistive-text/dist/index.js",
+      "maxSize": "2.75 KB"
```
Comment (ashleynolan) [inline bundlewatch.config.json:5]: Super NIT – I think this should be more like 1.3Kb looking at the report?

File: bundlewatch.config.json [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+      "path": "./packages/components/pie-textarea/dist/index.js",
+      "maxSize": "5 KB"
+    },
+    {
+      "path": "./packages/components/pie-textarea/dist/react.js",
+      "maxSize": "0.38 KB"
+    },
+    {
+      "path": "./packages/components/pie-toast/dist/index.js",
+      "maxSize": "6.25 KB"
```
Comment (ashleynolan) [inline bundlewatch.config.json:181]: and this one about 4.5kb

File: bundlewatch.config.json (hunk 4 above)
Comment (jamieomaguire) [reply bundlewatch.config.json]: pushed

File: bundlewatch.config.json (hunk 5 above)
Comment (jamieomaguire) [reply bundlewatch.config.json]: pushed

**Review by ashleynolan — APPROVED** [review]

**Pull request merged**